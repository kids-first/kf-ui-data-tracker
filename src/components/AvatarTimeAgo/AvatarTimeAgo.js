import React from 'react';
import PropTypes from 'prop-types';
import {Label} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
/**
 * Displays time ago and avatar in one label
 */
const AvatarTimeAgo = ({showUsername, creator, createdAt, size}) => {
  if (showUsername) {
    return (
      <Label image size={size}>
        {creator && (
          <img
            alt={creator.username || 'unknown'}
            src={
              creator.picture || 'https://www.w3schools.com/css/img_avatar.png'
            }
          />
        )}
        {creator && showUsername && creator.username}
        <Label.Detail>
          {createdAt ? <TimeAgo date={createdAt} live={false} /> : 'Unknown'}
        </Label.Detail>
      </Label>
    );
  } else {
    return (
      <Label image basic size={size}>
        {creator && (
          <img
            alt={creator.username || 'unknown'}
            src={
              creator.picture || 'https://www.w3schools.com/css/img_avatar.png'
            }
          />
        )}
        {createdAt ? <TimeAgo date={createdAt} live={false} /> : 'Unknown'}
      </Label>
    );
  }
};

AvatarTimeAgo.propTypes = {
  /** User object contains username and picture */
  creator: PropTypes.object,
  /** If to show user name */
  showUsername: PropTypes.bool,
  /** Show as time ago */
  createdAt: PropTypes.string,
  /** Label size. */
  size: PropTypes.oneOf([
    'mini',
    'tiny',
    'small',
    'medium',
    'large',
    'big',
    'huge',
    'massive',
  ]),
};

export default AvatarTimeAgo;
