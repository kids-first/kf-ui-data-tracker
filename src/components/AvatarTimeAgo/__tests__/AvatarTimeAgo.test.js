import React from 'react';
import {render, cleanup} from 'react-testing-library';
import AvatarTimeAgo from '../AvatarTimeAgo';

afterEach(cleanup);
it('renders AvatarTimeAgo showing username', async () => {
  const creator = {
    username: 'username',
    picture: 'https://www.w3schools.com/w3images/avatar6.png',
  };
  const createdAt = '2019-04-17T16:39:34+00:00';
  const {container} = render(
    <AvatarTimeAgo
      size="tiny"
      showUsername
      creator={creator}
      createdAt={createdAt}
    />,
  );
  expect(container).toMatchSnapshot();
});

it('renders AvatarTimeAgo not showing username', async () => {
  const creator = {
    username: 'username',
    picture: 'https://www.w3schools.com/w3images/avatar6.png',
  };
  const createdAt = '2019-04-17T16:39:34+00:00';
  const {container} = render(
    <AvatarTimeAgo size="mini" creator={creator} createdAt={createdAt} />,
  );
  expect(container).toMatchSnapshot();
});
