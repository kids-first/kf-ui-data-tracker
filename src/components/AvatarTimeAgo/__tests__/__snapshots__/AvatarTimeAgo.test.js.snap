// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AvatarTimeAgo not showing username 1`] = `
<div>
  <div
    class="ui mini basic image label"
  >
    <img
      alt="username"
      src="https://www.w3schools.com/w3images/avatar6.png"
    />
    <time
      datetime="2019-04-17T16:39:34.000Z"
      title="2019-04-17T16:39:34+00:00"
    >
      6 days ago
    </time>
  </div>
</div>
`;

exports[`renders AvatarTimeAgo showing username 1`] = `
<div>
  <div
    class="ui tiny image label"
  >
    <img
      alt="username"
      src="https://www.w3schools.com/w3images/avatar6.png"
    />
    username
    <div
      class="detail"
    >
      <time
        datetime="2019-04-17T16:39:34.000Z"
        title="2019-04-17T16:39:34+00:00"
      >
        6 days ago
      </time>
    </div>
  </div>
</div>
`;
